python/src/ir.cc

```diff
@@ -500,6 +509,55 @@ void init_triton_ir(py::module &&m) {
            [](ModuleOp &self, std::string &funcName) -> FuncOp {
              return self.lookupSymbol<FuncOp>(funcName);
            })
+      /*
+       * def ty_to_cpp(ty) is the consumer of this function.
+       * If the type is a ptr it expects ty[0] == '*', else the type itself.
+       */
+
+      .def("get_function_signature",
+           [](ModuleOp &self, FuncOp &func) -> std::vector<std::string> {
+             std::vector<std::string> strVec;
+
+             auto findTMA = [](ArrayRef<NamedAttribute> dictVals) {
+               for (auto attr : dictVals) {
+                 if (auto intAttr = dyn_cast<IntegerAttr>(attr.getValue())) {
+                   if (attr.getName() == "tt.nv_tma_desc" &&
+                       intAttr.getInt() == 1)
+                     return true;
+                 }
+               }
+               return false;
+             };
+
+             auto type = func.getFunctionType();
+             unsigned numArgs = type.getNumInputs();
+             for (unsigned i = 0; i != numArgs; ++i) {
+               std::string tempType;
+               llvm::raw_string_ostream os(tempType);
+
+               auto ty = type.getInput(i);
+               if (auto attributes = func.getCallableArgAttrs()) {
+                 Attribute attr = attributes[i];
+                 // Check for tt.nv_tma_desc = 1
+                 if (auto dAttr = dyn_cast<DictionaryAttr>(attr)) {
+                   ArrayRef<NamedAttribute> dictVals = dAttr.getValue();
+                   if (findTMA(dictVals)) {
```

Comment:
Could this not be
```suggestion
                   if (dAttr.contains("tt.nv_tma_desc")) {
```

Reply:
yeah it could be, it changes the behavior slightly as this just checks for any value but I'm fine with that (will adjust the PR in a sec w/fixed test)